lib/spack/spack/cmd/load.py

```diff
@@ -99,8 +100,18 @@ def load(parser, args):
 
     with spack.store.STORE.db.read_transaction():
         env_mod = uenv.environment_modifications_for_specs(*specs)
+        shell = args.shell if args.shell else os.environ.get("SPACK_SHELL")
         for spec in specs:
+            # if file cached shell script exists skip
+            spec_mod = uenv.environment_modifications_for_specs(*[spec])
+            spec_mod.prepend_path(uenv.spack_loaded_hashes_var, spec.dag_hash())
+            spec_cmds = spec_mod.shell_modifications(shell)
+
             env_mod.prepend_path(uenv.spack_loaded_hashes_var, spec.dag_hash())
-        cmds = env_mod.shell_modifications(args.shell)
+            cmds = env_mod.shell_modifications(shell)
+
+            path = os.path.join(spec.prefix, ".spack", f"{spec.name}_shell.{shell}")
```

Comment:
Users might not have access to write to this location in all use cases? If this is the approach, would this be better accomplished at install-time?

Reply:
Any package the users want to load must be installed first so I don't think this will cause an issue.

Reply:
Sorry if I'm being unclear - here's an example from our environment.
```
[aweits@gung ~]$ spack location -i gmake/s257rho
/.autofs/tools/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-7.4.0/gmake-4.2.1-s257rho7ws2xht2yw3jx3pud73em7dbq
spack:/tools/spack
[aweits@gung ~]$ ls -la /.autofs/tools/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-7.4.0/gmake-4.2.1-s257rho7ws2xht2yw3jx3pud73em7dbq/.spack
total 108
drwxr-sr-x 3 jchits staff  183662 Feb 12  2020 .
drwxr-sr-x 6 jchits staff 2541969 Feb 12  2020 ..
-rw-r--r-- 1 jchits staff   38562 Feb 12  2020 install_manifest.json
drwxr-sr-x 3 jchits staff   73780 Feb 12  2020 repos
-rw-r--r-- 1 jchits staff   14264 Feb 12  2020 spack-build-env.txt
-rw-r--r-- 1 jchits staff   46879 Feb 12  2020 spack-build-out.txt
-rw-r--r-- 1 jchits staff   10177 Feb 12  2020 spec.yaml
spack:/tools/spack
[aweits@gung ~]$ touch /.autofs/tools/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-7.4.0/gmake-4.2.1-s257rho7ws2xht2yw3jx3pud73em7dbq/.spack/foo
touch: cannot touch ‘/.autofs/tools/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-7.4.0/gmake-4.2.1-s257rho7ws2xht2yw3jx3pud73em7dbq/.spack/foo’: Permission denied
```

Reply:
Since we're going to change this to cache at install time I think this will work itself out by the end of the PR